xnat_tools/run_heudiconv.py

```diff
@@ -71,33 +74,81 @@ def run_heudiconv(
     )
 
     export_dir = f"{bids_root_dir}/{pi_prefix}/{study_prefix}/xnat-export"
-    dicom_dir_template = f"{export_dir}/{subject_prefix}/{session_prefix}"
+    dicom_dir = Path(export_dir) / subject_prefix / session_prefix
 
     # check if the extension of the images is dcm or IMA
-    dicom_ext = "dcm"
-    if len(glob.glob(f"{dicom_dir_template}/*/*.dcm")) == 0:
-        if len(glob.glob(f"{dicom_dir_template}/*/*.IMA")) > 0:
-            dicom_ext = "IMA"
+    if (len(glob.glob(f"{dicom_dir}/*/*.dcm")) == 0) and len(
+        glob.glob(f"{dicom_dir}/*/*.IMA")
+    ) == 0:
+        raise ValueError(f"No .dcm or .IMA files found in {dicom_dir}")
+
+    # all first-level series directories under the session
+    series_dirs = [d for d in dicom_dir.iterdir() if d.is_dir()]
+
+    # filter specific scan types / scanner derivatives that we know will break heudiconv
+    # some of these we will handle ourselves afterward
+    heudi_skip_list = [
+        "_MPR_",
+        "_ND",
+        "rf_map",
+        "tb1tfl",
+        "TB1TFL",
+        "_TRACEW",
+        "_TENSOR",
+        "_B0",
+        "_ColFA",
+        "_ADC",
+    ]
+
+    # split off bottom 3 levels because they need to be supplied as a locator path
+    heudi_base_dir = Path(heudi_output_dir).parents[3]
+    heudi_locator = Path(heudi_output_dir).relative_to(heudi_base_dir)
+
+    keep_dirs, skip_dirs = [], []
+
+    for d in series_dirs:
+        if any(s in d.name for s in heudi_skip_list):
+            skip_dirs.append(d)
         else:
-            raise ValueError(f"No .dcm or .IMA files found in {dicom_dir_template}")
-
-    heudi_cmd = f"heudiconv -f reproin --bids \
-    -o {heudi_output_dir} \
-    --dicom_dir_template {export_dir}/sub-{{subject}}/ses-{{session}}/*/*.{dicom_ext} \
-    --subjects {subject} --ses {session_suffix}"
+            keep_dirs.append(d)
+
+    print(f"Skipping heudiconv BIDS conversion for {', '.join(s.name for s in skip_dirs)}")
```

Comment:
Minor change. I might guard the print here with `if skip_dirs:` to prevent unnecessary prints to stdout.